### PR TITLE
Don't discard command output until we know we don't need it.

### PR DESCRIPTION
**Pull request by gbtitus**
We got 'IndexError: string index out of range' from python in
run_command() because we tried to print what the target command had
produced to stderr, but our caller had told us not to collect that.
The details were: CHPL_HOST_PLATFORM was set to hpe-cray-ex and
CHPL_LIBFABRIC was set to system, but there was no libfabric module
loaded. Thus the 'pkg-config --exists libfabric' command run from
pkgconfig_get_compile_args() for a 'printchplenv --internal' failed,
and then we tried to print stderr.  But we hadn't asked for it.

Here, adjust: always collect stdout and stderr but only return to the
user the one(s) they asked for.  (I started to just collect stderr
always, since that's the only one run_command() itself might need,
but that actually ends up requiring even more control flow.)

Signed-off-by: Greg Titus <gbtitus@users.noreply.github.com>

### CODE REVIEW

File: util/chplenv/utils.py [hunk 1]
```diff
@@ -62,14 +62,22 @@ def try_run_command(command, stdout=True, stderr=False, cmd_input=None):
 def run_command(command, stdout=True, stderr=False, cmd_input=None):
     """Command subprocess wrapper.
        This is the usual way to run a command and collect its output."""
-    exists, returncode, output = try_run_command(command, stdout, stderr,
-                                                 cmd_input)
+    exists, returncode, myOutput = try_run_command(command, True, True,
```
Comment (ben-albrecht) [inline util/chplenv/utils.py:65]: nit: python uses snake_case, e.g.

```suggestion
    exists, returncode, my_output = try_run_command(command, True, True,
```

**Review by ben-albrecht — APPROVED** [review]
Looks good overall.

From what I can tell, there are only 2 calls to `try_run_command` in all of `utils/`, one is here, and the other ignores the `output` return entirely. This could be an opportunity to simplify `try_run_command` interface to just always return `stdout` and `stderr`. I'd also be fine leaving it as is though.

**Comment by gbtitus** [comment]
> This could be an opportunity to simplify `try_run_command` interface to just always return `stdout` and `stderr`.

Yeah, that occurred to me after the fact also.  I'll take a look.

**Comment by gbtitus** [comment]
@ben-albrecht take another look?

File: util/chplenv/utils.py [hunk 2]
```diff
@@ -45,37 +45,30 @@ def try_run_command(command, stdout=True, stderr=False, cmd_input=None):
                                    stderr=subprocess.PIPE,
                                    stdin=subprocess.PIPE)
     except OSError:
-        return False, 0, ''
+        return False, 0, None, None
```
Comment (ben-albrecht) [inline util/chplenv/utils.py:48]: If I were writing this, I'd make this return a tuple explicitly to be consistent with the other return:

```suggestion
        return (False, 0, None, None)
```

**Review by ben-albrecht — APPROVED** [review]
Looks good, thanks Greg!

**Comment by gbtitus** [comment]
Yeah, saw that and followed the directions, so it's all repaired now.

**Review by ben-albrecht — APPROVED** [review]

**Pull request merged**